nengo/networks/tests/test_linear_system.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+import pytest
+
+import nengo
+from nengo.linear_system import LinearSystem
+from nengo.networks.linear_system import LinearSystemNetwork, ss2sim
+import nengo.utils.numpy as npext
+
+
+class LinearSystemNodeNetwork(LinearSystemNetwork):
+    def make_state_object(self, n_neurons, dimensions, **kwargs):
+        if "radius" in kwargs:
+            kwargs.pop("radius")
+        state = nengo.Node(size_in=dimensions, **kwargs)
+        return state, state, state
+
+
+class LinearSystemSReluNetwork(LinearSystemNetwork):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, neuron_type=nengo.SpikingRectifiedLinear(), **kwargs)
+
+
+def _test_linear_system_network(
+    sys,
+    synapse,
+    NetworkType,
+    Simulator,
+    seed=0,
+    plt=None,
+    dt=None,
+    simtime=1.0,
+    input_f=None,
+    **kwargs
+):
+    probe_synapse = nengo.Alpha(0.005)
+
+    with nengo.Network(seed=seed) as net:
+        ref = nengo.Node(sys, label="ref")
+        ref_p = nengo.Probe(ref, synapse=probe_synapse)
```

Comment:
There's also a way to do this without probing `ref` with a synapse. You can take `subnet.state_input` which is `x` and then do `Cx + Du` directly to get the network's output. The `subnet.state_input` is already filtered and so no additional filtering is required on either the reference or the network. This is a feature that I've wanted a number of times in nengolib but was just doing it manually when needed. If you think it's a good idea I think this would be a nice add-on to `LinearNetwork`, possibly as an alternative output node...? The main drawback of this approach is if `u` is spiking then it won't look great.

Reply:
So it would be an alternative output node that uses `state_input` instead of `state_output` as the `pre` for the "C" connection? That seems like a reasonable thing to want to do sometimes, but the fact that the connection doesn't get represented using neurons is somewhat concerning. I would leave this as something users can do themselves if they wish (it seems like more of a power-user move, and only requires a line or two of code).

Reply:
Yeah it is a bit of a power move. And can also mean some interesting things sometimes, like the `Cx` connection that doesn't get represented in neurons could be multiplied together with the connection that it's eventually projecting to. And if there is a lowpass synapse on that connection too, then it becomes an `DoubleExp` and can give you better signal to noise. It would be nice if these nuances were discoverable, as it's one of those things that's not readily obvious, but can become clear and understandable with an example, IMO. 

It can be done by the user in a line or two of code, but requires the user to look into the code and pull out `C, D = subnet.mapped_system.ss[2:]` to build the two connections, as well as realize that each connection needs to go from `subnet.state.input`. To avoid having to know these internals, there could be a method like `add_output` (akin to the `EnsembleArray`) method. But I am wary of wanting to add more to this already large PR, so we could also move this discussion to a later feature request issue.

Reply:
I also realized you can do this same thing but from the output of the `EnsembleArray`, which is equivalent to what I was saying above but is perhaps more conventional since we're no longer connecting from the input to the array. So I added a commit which shows how to do this in the unit test. It was actually pretty nontrivial to do this (took me longer than I thought because there are some nuances to initializing the synapse). Hoping this can serve as a reference if in the future someone wants to apply the same trick or if there is some desired to turn this into a helper method.